media_driver/agnostic/media_srcs.cmake

```diff
@@ -18,6 +18,79 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 
+if (BUILD_KERNELS)
+    # Here we define build steps to generate c-array binary shaders (kernels)
+    # and their header files. If you don't use BUILD_KERNELS option these
+    # kernels will just be used from pre-built form. If you will regenerate
+    # kernels you may notice the difference from the per-built kernels in git-diff.
+
+    # iga64 -i <src> -o <dst> -p <genx> -a
+    function(iga_compile src dst genx)
+        get_filename_component(tgt ${src} NAME)
+        add_custom_command(
+          OUTPUT ${dst}
+          DEPENDS KrnToHex_IGA ${src}
+          COMMAND ${IGA} -p ${genx} -a ${src} -o ${tgt}.krn
+          COMMAND KrnToHex_IGA ${tgt}.krn
+          COMMAND ${CMAKE_COMMAND} -E remove ${tgt}.krn
+          COMMAND ${CMAKE_COMMAND} -E rename ${tgt}.hex ${dst})
+    endfunction()
+
+    function(platform_to_genx platform genx kind)
+        if(platform STREQUAL "gen11_icllp")
+            set(genx "11" PARENT_SCOPE)
+            set(kind "icllp" PARENT_SCOPE)
+        endif()
+    endfunction()
+
+    # Function generate kernel for the specified platform. It assumes
+    # that kernel sources are located in a certain location (see ${krn_dir})
+    # calculated in a function body. And it assumes that output files should
+    # be stored in a certain locations (see ${krn_dir}/${krn}.h/c and ${krn_header}).
+    # Notes:
+    #   - <platform> = gen<X>_<kind>, for example, gen11_icllp
+    # TODO:
+    #   - As of now this function works for the single platform only (gen11_icllp) while it
+    #  actully should process all the platforms at once. This single processing
+    #  is needed to generate shared ${krn_header} with the unified kernels definition.
+    function(gen_kernel_from_asm name platform)
+        platform_to_genx(${platform} genx kind)
+
+        set(krn_dir ${CMAKE_SOURCE_DIR}/media_driver/agnostic/${platform}/vp/kernel)
```

Comment:
It is not a good idea to set new generated files under our repo. it is better to select cmak out put folder as new generated files destination folder